frontend/src/app/choreography/choreography.component.ts

```diff
@@ -498,4 +500,22 @@ export class ChoreographyComponent {
     }
     this.activeChoreographyItem = item;
   }
+
+  saveToDatabase(): void {
+    const dialogRef = this.dialog.open(SaveChoreographyDialogComponent, {});
+    dialogRef
+      .afterClosed()
+      .pipe(filter((result) => !!result))
```

Comment:
**question:** what's the real return type? `!!` is a bit of a hack to make something truthy whereas we would like to check the actual value

Reply:
afterClosed() returns `Observable<R | undefined>`. It returns 'true' when the user clicks 'ok' in the dialog, 'false' when the user clicks 'cancel' in the dialog and 'undefined' when the user clicks outside the dialog box.

Reply:
I guess it would work then without the double exclamation mark as undefined and false are falsy. In theory `result === true` would be the purest solution I guess.

Reply:
Changed to `result !== true`.